srm.js

```diff
@@ -25,7 +25,36 @@ function rethrowError(err) {
     throw parseError(err)
 }
 
-class CodeDxApiClient {
+function checkIfBranchPresent(branches, branchName) {
+    let result = false
+    branches.forEach(branch => {
+        if (branch.name.includes(branchName)) {
+            result = true
+        }
+    })
+    return result
+}
+
+function buildQueryParams(branches, baseBranchName, targetBranchName) {
+    const baseBranchPresent = checkIfBranchPresent(branches, baseBranchName)
+    const targetBranchPresent = checkIfBranchPresent(branches, targetBranchName)
+
+    // If base branch is not defined, OR, base branch is present but target branch isn't, add the base branch name in the query param
+    const baseBranchPart = ((typeof baseBranchName == 'undefined') || (baseBranchPresent && targetBranchPresent)) ? "" : `;branch=${baseBranchName}`
+    // If target branch is not defined, no need to add the branch name to the query param. It will work on the default branch
+    const targetBranchPart = (typeof targetBranchName == 'undefined') ? "analysis" : `analysis?branchName=${targetBranchName}`
+
+    return baseBranchPart + "/" + targetBranchPart
```

Comment:
I was confused by the use of `/` here, but I see this also builds part of the project URL when the name made me think it was purely for query params

I think this logic would make more sense if it was moved directly into `runAnalysis`, where the use of `baseBranch` as part of the project ID could be done directly, and you could simplify by replacing the `checkIfBranchPresent(branches, someBranchName)` checks with `!!someBranchName` since this logic will only be called after we've already validated the selections

(You could also do those checks by just using `someBranchName` directly, e.g. `(baseBranchName && targetBranchName)`, but use `!!someBranchName` explicitly converts the value to a `boolean`. The final logic will work out the same either way, so not requiring that you use `!!someBranchName`, but just FYI)